### PR TITLE
fix(redis): cluster list crash

### LINKED ISSUE #3685: redis: cluster list crash
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

## Command attempted

```
> scw redis cluster list
```

### Expected Behavior

List to work

### Actual Behavior

```
---------------------------------------------------------------------------------------
An error occurred, we are sorry, please consider opening a ticket on github using: 'scw feedback bug'
Give us as many details as possible so we can reproduce the error and fix it.
---------------------------------------------------------------------------------------
runtime error: invalid memory address or nil pointer dereference
stacktrace from panic: 
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
main.cleanup(0xc00027f0a0)
	/home/julescasteran/Documents/devtools/scaleway-cli/cmd/scw/main.go:42 +0xca
panic({0x1745400?, 0x2819370?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/scaleway/scaleway-cli/v2/internal/namespaces/redis/v1.redisEndpointsClusterGetMarshalerFunc({0x1694e60?, 0xc0009a2258?}, 0x0)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/namespaces/redis/v1/custom_cluster.go:165 +0x15e
github.com/scaleway/scaleway-cli/v2/internal/human.marshalInlineSlice({0x1694e60, 0xc0009a2258, 0x97})
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/human/marshal.go:359 +0x239
github.com/scaleway/scaleway-cli/v2/internal/human.marshalSlice({0x1694f20?, 0xc0009a2240?, 0x1761d00?}, 0xc0008279a0)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/human/marshal.go:326 +0x836
github.com/scaleway/scaleway-cli/v2/internal/human.Marshal({0x1694f20, 0xc0009a2240}, 0xc0009a2240?)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/human/marshal.go:81 +0x35e
github.com/scaleway/scaleway-cli/v2/internal/core.(*Printer).printHuman(0xc0000414a0, {0x1694f20, 0xc0009a2240}, 0x0)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/core/printer.go:202 +0x198
github.com/scaleway/scaleway-cli/v2/internal/core.(*Printer).Print(0xc0000414a0, {0x1694f20?, 0xc0009a2240?}, 0x199ec87?)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/core/printer.go:159 +0xef
github.com/scaleway/scaleway-cli/v2/internal/core.Bootstrap(0xc000b07e78)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/core/bootstrap.go:281 +0x156a
main.main()
	/home/julescasteran/Documents/devtools/scaleway-cli/cmd/scw/main.go:74 +0x33f


```

## More info

Based on master (f88725888bf6d0c41cca2f3893c05e4852b5954d)
Here is json output:
```
[{"id":"9413e3dd-2a46-4b6c-a919-ac4810d610ed","name":"test_redis_basic","project_id":"105bdce1-64c0-48ab-899d-868455867ecf","status":"error","version":"7.0.5","endpoints":[{"port":6379,"public_network":{},"ips":["51.158.100.79"],"id":"62bb9b63-69f7-4701-8917-f99db6b99e86"}],"tags":["test1"],"node_type":"RED1-S","created_at":"2024-02-27T00:49:20.839198Z","updated_at":"2024-02-27T00:53:44.255673Z","tls_enabled":true,"cluster_settings":[],"acl_rules":[],"cluster_size":1,"zone":"fr-par-1","user_name":"my_initial_user","upgradable_versions":[]},{"id":"2941895e-2de3-4dd2-9001-3e101a81f37c","name":"test_redis_migrate_cluster_size_static","project_id":"105bdce1-64c0-48ab-899d-868455867ecf","status":"error","version":"7.0.5","endpoints":[{"port":6379,"private_network":{"id":"d0895cb1-52fc-4eea-8e59-d4c85c81390e","service_ips":["192.168.99.1/24","192.168.99.2/24","192.168.99.3/24","192.168.99.4/24","192.168.99.5/24"],"zone":"fr-par-1","provisioning_mode":"static"},"ips":["192.168.99.1","192.168.99.2","192.168.99.3","192.168.99.4","192.168.99.5"],"id":"1fccc886-2e4e-4c6b-ac17-c6cfd9b5e89b"}],"tags":[],"node_type":"RED1-XS","created_at":"2024-02-28T00:54:21.16868Z","updated_at":"2024-02-28T01:08:47.732315Z","tls_enabled":true,"cluster_settings":[],"acl_rules":[],"cluster_size":5,"zone":"fr-par-1","user_name":"my_initial_user","upgradable_versions":[]}]
```

### PR DESCRIPTION
**Pull request by Laure-di**
close #3685

### CODE REVIEW

**Comment by Codelax** [comment]
Still crash in another case
```
❯ scw redis endpoint get 18a0f371-0fa7-412d-a73d-ea85cb96ae3a   
---------------------------------------------------------------------------------------
An error occurred, we are sorry, please consider opening a ticket on github using: 'scw feedback bug'
Give us as many details as possible so we can reproduce the error and fix it.
---------------------------------------------------------------------------------------
interface conversion: interface {} is redis.Endpoint, not []*redis.Endpoint
stacktrace from panic: 
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
main.cleanup(0xc0002d8460)
	/home/julescasteran/Documents/devtools/scaleway-cli/cmd/scw/main.go:42 +0xca
panic({0x17620c0?, 0xc000727230?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/scaleway/scaleway-cli/v2/internal/namespaces/redis/v1.redisEndpointsClusterGetMarshalerFunc({0x18bdde0?, 0xc0000d3240?}, 0x18bdde0?)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/namespaces/redis/v1/custom_cluster.go:164 +0x1db
github.com/scaleway/scaleway-cli/v2/internal/human.Marshal({0x18bdde0, 0xc0000d3240}, 0x1761d40?)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/human/marshal.go:61 +0x4db
github.com/scaleway/scaleway-cli/v2/internal/human.Marshal({0x16717c0, 0xc000677000}, 0xc000677000?)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/human/marshal.go:77 +0x392
github.com/scaleway/scaleway-cli/v2/internal/core.(*Printer).printHuman(0xc00015ccc0, {0x16717c0, 0xc000677000}, 0x0)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/core/printer.go:202 +0x198
github.com/scaleway/scaleway-cli/v2/internal/core.(*Printer).Print(0xc00015ccc0, {0x16717c0?, 0xc000677000?}, 0x199edc7?)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/core/printer.go:159 +0xef
github.com/scaleway/scaleway-cli/v2/internal/core.Bootstrap(0xc000b91e78)
	/home/julescasteran/Documents/devtools/scaleway-cli/internal/core/bootstrap.go:281 +0x156a
main.main()
	/home/julescasteran/Documents/devtools/scaleway-cli/cmd/scw/main.go:74 +0x33f


```

File: internal/namespaces/redis/v1/custom_cluster.go [hunk 1]
```diff
@@ -160,8 +160,9 @@ func ACLAddListBuilder(c *core.Command) *core.Command {
 }
 
 func redisEndpointsClusterGetMarshalerFunc(i interface{}, opt *human.MarshalOpt) (string, error) {
-	type tmp []*redis.Endpoint
-	redisEndpointsClusterResponse := tmp(i.([]*redis.Endpoint))
+	type tmp redis.Endpoint
+	redisEndpointsClusterResponse := tmp(i.(redis.Endpoint))
+	log.Println("value of redisEndpointsClusterResponse: ", redisEndpointsClusterResponse)
```
Comment (Codelax) [inline internal/namespaces/redis/v1/custom_cluster.go:165]:
```suggestion
```

File: internal/namespaces/redis/v1/custom_cluster.go [hunk 2]
```diff
@@ -160,8 +160,9 @@ func ACLAddListBuilder(c *core.Command) *core.Command {
 }
 
 func redisEndpointsClusterGetMarshalerFunc(i interface{}, opt *human.MarshalOpt) (string, error) {
-	type tmp []*redis.Endpoint
-	redisEndpointsClusterResponse := tmp(i.([]*redis.Endpoint))
+	type tmp redis.Endpoint
+	redisEndpointsClusterResponse := tmp(i.(redis.Endpoint))
```
Comment (Codelax) [inline internal/namespaces/redis/v1/custom_cluster.go:164]:
```suggestion
	endpoint := tmp(i.(redis.Endpoint))
```

File: internal/namespaces/redis/v1/custom.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ func GetCommands() *core.Commands {
 	cmds := GetGeneratedCommands()
 
 	human.RegisterMarshalerFunc(redis.Cluster{}, redisClusterGetMarshalerFunc)
-	human.RegisterMarshalerFunc(redis.Cluster{}.Endpoints, redisEndpointsClusterGetMarshalerFunc)
+	human.RegisterMarshalerFunc(redis.Endpoint{}, redisEndpointsClusterGetMarshalerFunc)
```
Comment (Codelax) [inline internal/namespaces/redis/v1/custom.go:13]: Should this marshaler affect the human output of `scw redis endpoint get` ?
```
❯ scw redis endpoint get 18a0f371-0fa7-412d-a73d-ea85cb96ae3a
Port  6379
IPs   [163.172.178.253]
ID    18a0f371-0fa7-412d-a73d-ea85cb96ae3a
```

File: internal/namespaces/redis/v1/custom_cluster.go [hunk 4]
```diff
@@ -160,8 +159,11 @@ func ACLAddListBuilder(c *core.Command) *core.Command {
 }
 
 func redisEndpointsClusterGetMarshalerFunc(i interface{}, opt *human.MarshalOpt) (string, error) {
+	if opt == nil {
+		return "", nil
+	}
```
Comment (Codelax) [inline internal/namespaces/redis/v1/custom_cluster.go:164]: It breaks output in cluster listing
```
❯ scw redis cluster list -ohuman=ID,Endpoints,NodeType
ID                                    ENDPOINTS  NODE TYPE
2b24ffd8-3408-4b1a-beea-ab8bb99207a9             RED1-micro
```

File: internal/namespaces/redis/v1/custom.go [hunk 5]
```diff
@@ -10,7 +10,7 @@ func GetCommands() *core.Commands {
 	cmds := GetGeneratedCommands()
 
 	human.RegisterMarshalerFunc(redis.Cluster{}, redisClusterGetMarshalerFunc)
-	human.RegisterMarshalerFunc(redis.Cluster{}.Endpoints, redisEndpointsClusterGetMarshalerFunc)
+	human.RegisterMarshalerFunc(redis.Cluster{}, redisEndpointsClusterGetMarshalerFunc)
```
Comment (Codelax) [inline internal/namespaces/redis/v1/custom.go:13]: This add two marshalers for the same type

Comment (Codelax) [reply internal/namespaces/redis/v1/custom.go:13]:
```
❯ scw redis cluster get 2b24ffd8-3408-4b1a-beea-ab8bb99207a9 
ID                                    Port  IPs
18a0f371-0fa7-412d-a73d-ea85cb96ae3a  6379  [163.172.178.253]
```

**Review by remyleone — APPROVED** [review]

**Pull request merged**